File: apps/backend/app/schemas/file.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class ProjectFileResponse(BaseModel):
    id: UUID
    project_id: UUID
    file_type: str
    original_filename: str
    mime_type: str | None
    file_size: int | None
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ProjectFileListResponse(BaseModel):
    items: list[ProjectFileResponse]

File: apps/frontend/types/file.ts
export type ProjectFile = {
  id: string;
  project_id: string;
  file_type: string;
  original_filename: string;
  mime_type: string | null;
  file_size: number | null;
  status: string;
  created_at: string;
};

export type ProjectFileListResponse = ProjectFile[];
